[Feature] First draft of language list

rvyuha opened the pull request.

yulric commented on qmd/languages.qmd:
```diff
@@ -1,3 +1,77 @@
+  # Check for not empty values
+  # @RUSTY consider functioning out checks for length
+  if (length(nat_name) < 1) {
+    nat_name <- "Native Name is missing at the momment"
```
These should all be wrapped in functions and put next to the initialized variables

rvyuha commented:
Updates since last review:
-Swapped over to direct excel usage.
-Additional checks with format_table.
-Verified every display input.
-Addressed PR comment with renaming of working_row to languages_table_row.

rvyuha commented:
Potentially lacking features due to the spec being created before their usage:
- Anchoring to language ID.
- Linking language ID to appropriate parts.qmd section. 
- Replacing h1 with h2

rvyuha commented:
Warning the latest dictionary version contains a bug in the languages sheet. It was manually fixed on this branch but should be considered prior to merging.

DougManuel reviewed (approved):
We'll need to have @mathew-thomson review, but the implementation looks good. The actual list is short with only three languages. 

I suggest that we merge to main, but don't add the actual page to the public page -- meaning, don't add to _qaurto.yaml until we review the content further.

yulric commented on R/format-table.R:
```diff
@@ -11,7 +11,8 @@ source(file.path(getwd(), "R", "constants.R"))
   function(input_table,
            columns_to_format = NULL,
-           remove_duplicate = FALSE) {
+           remove_duplicate = FALSE,
```
Can you documentation for two params `remove_duplicate` and `strip_invalid_part_ID`.

yulric commented on R/format-table.R:
```diff
@@ -11,7 +11,8 @@ source(file.path(getwd(), "R", "constants.R"))
 format_table <-
```
Can you leave function descriptions for the `remove_duplicate` and `strip_invalid_part_ID` parameters?

DougManuel commented:
In retrospect, we should now create a dev branch and should have PR to dev as opposed to main. As well, I would like to start versioning our documentation. We can discuss at our in-person meeting.

DougManuel commented:
@mathew-thomson Could you add to the introduction of languages.qmd? Then I think the page can go live on main.

rvyuha commented:
Restored branch since I am unsure about its use currently. I believe creating new branch for next steps of language page is the way to go. Let me know if I can proceed with deleting this branch.
